cli/cmd/service.go

```diff
@@ -1315,19 +1318,25 @@ func (c *ServicedCli) cmdServiceListSnapshots(ctx *cli.Context) {
 
 // serviced service snapshot SERVICEID
 func (c *ServicedCli) cmdServiceSnapshot(ctx *cli.Context) {
-	if len(ctx.Args()) < 1 {
+	nArgs := len(ctx.Args())
+	if nArgs < 1 {
 		fmt.Printf("Incorrect Usage.\n\n")
 		cli.ShowCommandHelp(ctx, "snapshot")
 		return
 	}
 
+	description := ""
+	if nArgs <= 3 {
```

Comment:
I wanted to treat the new arguments for `--description "your description here"` as optional. In other words, don't call `ctx.String("description")` unless those arguments were added.
Maybe the check should have been `if nargs == 3`?